lib/clusters.py

```diff
@@ -141,10 +144,21 @@ def update_clusters(all_clusters, trackings) -> None:
       cluster = Cluster(tracking.group)
       all_clusters.append(cluster)
 
+    # If we are adding a new tracking or order ID, unset the manual override
+    # status of the cluster.
+    override_overridden = False
+    if (len(set(tracking.order_ids).difference(set(cluster.orders))) > 0 or
```

Comment:
pretty sure the len() > 0 part can be removed, can just say "if set(tracking.order_ids).difference(set(cluster.orders)) or ..."

Reply:
I considered that option, but I'm increasingly hating Python's duck-typiness, and the fact that you can't tell what anything actually is just by looking at the callsite if all you're evaluating is truthiness, so I wanted to be very explicit that what's being checked is whether a collection contains any elements.  Just saying `if foo` doesn't give that context -- it's not even clear if foo is a collection at all.